src/cloud-cli/meltano/cloud/cli/project.py

```diff
@@ -0,0 +1,291 @@
+}
+
+
+@project_group.command("list")
+@click.option(
+    "--limit",
+    required=False,
+    type=int,
+    default=DEFAULT_GET_PROJECTS_LIMIT,
+    help="The maximum number of projects to display.",
+)
+@click.option(
+    "--format",
+    "output_format",
+    required=False,
+    default="terminal",
+    type=click.Choice(("terminal", "markdown", "json")),
+    help="The output format to use.",
+)
+@pass_context
+@run_async
+async def list_projects(
+    context: MeltanoCloudCLIContext,
+    output_format: str,
+    limit: int,
+) -> None:
+    """List Meltano Cloud projects."""
+    click.echo(
+        project_list_formatters[output_format](
+            [
+                _remove_private_project_attributes(x)
+                for x in (await _get_projects(config=context.config, limit=limit))
+            ],
+        ),
+    )
+
+
+def _check_for_duplicate_project_names(projects: list[CloudProject]) -> None:
+    project_names = [x["project_name"] for x in projects]
+    if len(set(project_names)) != len(project_names):
+        click.secho(
+            "Error: Multiple Meltano Cloud projects have the same name. "
+            "Please specify the project using the `--id` option with its "
+            "internal ID, shown below. Note that these IDs may change at any "
+            "time. To avoid this issue, please use unique project names.",
+            fg="red",
+        )
+        for project in projects:
+            click.echo(
+                f"{project['project_id']}: {project['project_name']} "
+                f"({project['git_repository']!r})",
+            )
+        sys.exit(1)
+
+
+class ProjectChoicesQuestionaryOption(click.Option):
+    """Click option that provides an interactive prompt for Cloud Project names."""
+
+    def prompt_for_value(self, ctx: click.Context) -> t.Any:
+        """Prompt the user to interactively select a Meltano Cloud project by name.
+
+        Args:
+            ctx: The Click context.
+
+        Returns:
+            The name of the selected project, or `None` if the project was
+            selected using the `--id` option.
+        """
+        if "project_id" in ctx.params:
+            # The project has been specified by ID - don't prompt for a name
+            return None
+        context: MeltanoCloudCLIContext = ctx.obj
+        context.projects = asyncio.run(_get_projects(context.config))
+        _check_for_duplicate_project_names(context.projects)
+        default_project_name = next(
+            (
+                x
+                for x in context.projects
+                if x["project_id"] == context.config.default_project_id
+            ),
+            {"project_name": None},
+        )["project_name"]
+        return questionary.select(
+            message="",
+            qmark="Use Meltano Cloud project",
+            choices=[x["project_name"] for x in context.projects],
+            default=default_project_name,
+        ).unsafe_ask()
```

Comment:
According to the docs this

<blockquote>

Does not catch keyboard interrupts.

</blockquote>

However, in my testing Cmd+C does seem to abort the prompt so I wonder if this is handled by Click.

Reply:
Yes, it is being handled by Click. I should add a comment to explain why I'm using `unsafe_ask`, since the name is pretty suspicious. If you use `ask` instead, it still aborts execution, but with a different message than what Click provides.